shell/common/engine.cc

```diff
@@ -234,11 +234,25 @@ void Engine::RunBundleAndSource(const std::string& bundle_path,
   std::string packages_path = packages;
   if (packages_path.empty())
     packages_path = FindPackagesPath(main);
-  if (!bundle_path.empty())
-    ConfigureAssetBundle(bundle_path);
-  ConfigureRuntime(GetScriptUriFromPath(main));
-  load_script_error_ =
-      runtime_->dart_controller()->RunFromSource(main, packages_path);
+
+  std::vector<uint8_t> platform_kernel;
+  if (!bundle_path.empty()) {
+    FXL_LOG(WARNING) << "RunBundleAndSource bundle_path is not empty";
+    GetAssetAsBuffer(blink::kPlatformKernelAssetKey, &platform_kernel);
+  }
+  ConfigureRuntime(GetScriptUriFromPath(bundle_path), platform_kernel);
+
+  if (!platform_kernel.empty()) {
```

Comment:
Instead of triggering this using the presence of platform_kernel should we not be using the --preview-dart-2 option as the way to trigger it. This would ensure that we don't inadvertently load stuff from source when --preview-dart-2 is used.

Reply:
Currently we don't propagate that flag down to the flutter engine, instead we use presence of platform kernel dill file as an indication that we should be running in kernel mode: https://github.com/flutter/engine/blob/master/runtime/dart_controller.cc#L178

Reply:
Until we link platform_kernel into the binary can we not at least remember the value of platform_kernel in the dart_controller object so that we don't have to include 'platform_kernel' for every hot restart.